### PR TITLE
Produce summary for unsupported features in top 100 crates experiment

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

No changes to Kani itself, just adds/changes scripts related to the top 100 crates experiment. In particular, it adds:
 1. A script to filter all unsupported constructs expressions. Before this change, the top-level script printed only the first 3 lines after a warning, which means the data was not complete.
 2. A script to produce a summary (table) from unsupported features data.

The top-level script is called the same way, but now it produces the following output at the end:
```
--- OVERALL STATS ---
10 crates failed to compile
33 crates had warning(s)

SUMMARY - UNSUPPORTED FEATURES
=========================================================
Unsupported feature | Crates impacted | Instances of use
---------------------------------------------------------
                      Rvalue::ThreadLocalRef  |  20 |  33
                         'simd_or' intrinsic  |  14 |  17
                         'simd_eq' intrinsic  |  11 |  13
                         'simd_gt' intrinsic  |  10 |  10
          Constant slice value with 2+ bytes  |   9 |  74
                   'simd_shuffle4' intrinsic  |   6 |  27
                        'simd_xor' intrinsic  |   6 |  12
                        'simd_and' intrinsic  |   6 |  11
                   'simd_shuffle8' intrinsic  |   5 |  20
                                         try  |   4 |  13
                  'simd_shuffle16' intrinsic  |   4 |  12
                   'simd_shuffle2' intrinsic  |   4 |   9
                        'simd_add' intrinsic  |   4 |   9
                     'simd_insert' intrinsic  |   3 |  11
                    'simd_extract' intrinsic  |   3 |   5
               PointerCast::ClosureFnPointer  |   2 |   6
                  'simd_shuffle32' intrinsic  |   1 |  34
  drop_in_place for drop_unimplemented__RINvN |   1 |   1
  drop_in_place for drop_unimplemented__RINvN |   1 |   1
  drop_in_place for drop_unimplemented__RINvN |   1 |   1
  drop_in_place for drop_unimplemented__RINvN |   1 |   1
  drop_in_place for drop_unimplemented__RINvN |   1 |   1
  Unsupported drop unsized struct: std::cell: |   1 |   1
                         'sqrtf64' intrinsic  |   1 |   1
                         'powif64' intrinsic  |   1 |   1
                         'powif32' intrinsic  |   1 |   1
                          'logf64' intrinsic  |   1 |   1
                          'expf64' intrinsic  |   1 |   1
=========================================================
```

### Testing:

* How is this change tested? Locally.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: scripts/exps/kani-repos-utils/filter_unsupported_constructs.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+if [[ $# -ne 1 ]]; then
+  echo "$0: Error: Missing input file."
+  exit 1
+fi
+
+# Filter lines with unsupported constructs.
```
Comment (celinval) [inline scripts/exps/kani-repos-utils/filter_unsupported_constructs.sh:10]: Nit: Move this doc to the top of the file.

File: scripts/exps/kani-repos-utils/filter_unsupported_constructs.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+if [[ $# -ne 1 ]]; then
+  echo "$0: Error: Missing input file."
```
Comment (celinval) [inline scripts/exps/kani-repos-utils/filter_unsupported_constructs.sh:6]: Can you make this error a bit clearer? What is the input file?

File: scripts/exps/kani-repos-utils/produce_unsupported_summary.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+if [[ $# -ne 1 ]]; then
+  echo "$0: Error: Missing input file."
+  exit 1
+fi
+
+# Print a table from a file with data about unsupported features.
```
Comment (celinval) [inline scripts/exps/kani-repos-utils/produce_unsupported_summary.sh:10]: Same. This is documenting the script and should be at the top.

File: scripts/exps/kani-repos-utils/produce_unsupported_summary.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#  2. The number of instances of a given construct
+#     (i.e., the number of instances according to the captured value)
+#
+# The data is sorted in descending order by the number of crates impacted.
+echo "SUMMARY - UNSUPPORTED FEATURES"
+echo "========================================================="
+echo "Unsupported feature | Crates impacted | Instances of use"
+echo "---------------------------------------------------------"
+cat $1 | sed -n 's/^ *- \([^(]*\)(\([0-9]*\))/\2 | \1/p' | \
+awk -F '|' '{
```
Comment (celinval) [inline scripts/exps/kani-repos-utils/produce_unsupported_summary.sh:30]: Should this be an awk script instead of bash?

**Review by celinval — COMMENTED** [review]
Is there a reason why we have these functionalities as separate scripts? Do we expect to run these separately?

I really wish we would use less bash scripts. They can get rather complicated.

File: scripts/exps/kani-repos-utils/filter_unsupported_constructs.sh (hunk 2 above)
Comment (adpaco-aws) [reply scripts/exps/kani-repos-utils/filter_unsupported_constructs.sh]: Added clarification to the error.

File: scripts/exps/kani-repos-utils/produce_unsupported_summary.sh (hunk 4 above)
Comment (adpaco-aws) [reply scripts/exps/kani-repos-utils/produce_unsupported_summary.sh:31]: Well, `sed` and `sort` are doing most of the work here. I think it's best to keep the `awk` program simple.

**Comment by adpaco-aws** [comment]
> Is there a reason why we have these functionalities as separate scripts? Do we expect to run these separately?
> 
> I really wish we would use less bash scripts. They can get rather complicated.

I put these as separate because otherwise I've to escape the regexes until they're no longer readable. I agree we should less bash scripts, and we intend to replace them with an actual tool soon. But in the meantime, these scripts will help us process data for unsupported features.

**Review by celinval — APPROVED** [review]

**Pull request merged**